### PR TITLE
Add custom params validation

### PR DESCRIPTION
**Pull request by tarmolov**
/cc @lemmy- @zloylos

### CODE REVIEW

**Comment by tarmolov** [comment]
🐖 🐖 🐖  🏠  🐺

File: REFERENCE.md [hunk 1]
```diff
@@ -116,8 +115,24 @@ var helloMethod = new ApiMethod({
...
+});
+```
+
+You can specify a function instead of object for a custom params validation. First argument is params which are passed to the method and the second one is Express request when method is executed from the client side.
+```javascript
+var helloMethod = new ApiMethod({
+    name: 'hello',
+    params: function (params, req) {
+        if (typeof params.name !== 'string') {
+            thrown new ApiError(ApiError.BAD_REQUST, 'Invalid username param');
```
Comment (lemmy-) [inline REFERENCE.md]: `throw`?

File: lib/api.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ var Api = inherit({
     /**
      * @param {String} methodPathPattern Path to api files. Path includes a file mask (supports minimatch).
      * @param {Object} [options] Api options.
-     * @param {String|Function} [options.paramsValidation=normalize] Preprocessing method parameters.
+     * @param {String} [options.paramsValidation=normalize] Preprocessing method parameters.
```
Comment (zloylos) [inline lib/api.js]: This options is really need?

Comment (tarmolov) [inline lib/api.js]: Yep. It's a way to change built-in validation.

File: REFERENCE.md (hunk 1 above)
Comment (tarmolov) [inline REFERENCE.md]: Thx. Fixed :)

**Comment by lemmy-** [comment]
🚢

**Comment by lemmy-** [comment]
Haven't we broken the docpage? [changes](https://github.com/baby-loris/bla/blob/018e482e2d51a5a4873d03f834f3c8f0eab11ede/lib/api-method.js#L126) vs [usage](https://github.com/baby-loris/bla/blob/018e482e2d51a5a4873d03f834f3c8f0eab11ede/lib/help-formatters/html.jade#L17
)

**Comment by tarmolov** [comment]
if a user creates a custom validator, docpage will work but no param will be shown.

**Comment by lemmy-** [comment]
Ok, if this behavior was implied.

**Pull request closed without merging**